### PR TITLE
Add Amazon team to recipients for builds they want.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
As discussed offline, since their team is taking responsibility for these builds, they'll get the emails associated with their status.

### CODE REVIEW

File: create_jenkins_job.py [hunk 1]
```diff
@@ -191,15 +191,15 @@ def create_job(os_name, job_name, template_file, additional_dict):
                 'cmake_build_type': 'Debug',
                 'compile_with_clang_default': 'true',
                 'time_trigger_spec': PERIODIC_JOB_SPEC,
-                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS + ',amazon-b9-ros@amazon.com',
```
Comment (dirk-thomas) [inline create_jenkins_job.py]: Are you sure they are comma separated? Other Jenkins jobs use a space to separate emails.

Comment (nuclearsandwich) [reply create_jenkins_job.py]: Ah no, hazarded a guess. Fixed in 6e02283.

File: create_jenkins_job.py [hunk 2]
```diff
@@ -191,15 +191,15 @@ def create_job(os_name, job_name, template_file, additional_dict):
                 'cmake_build_type': 'Debug',
                 'compile_with_clang_default': 'true',
                 'time_trigger_spec': PERIODIC_JOB_SPEC,
-                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS + ' amazon-b9-ros@amazon.com',
```
Comment (thomas-moulard) [inline create_jenkins_job.py]: ros-contributions@amazon.com would be better IMHO - is this email used somewhere else?

File: create_jenkins_job.py [hunk 3]
```diff
@@ -191,15 +191,15 @@ def create_job(os_name, job_name, template_file, additional_dict):
...
             })
 
         # configure nightly job for testing rmw/rcl based packages with thread sanitizer on linux
         if os_name == 'linux':
             create_job(os_name, 'nightly_' + os_name + '_thread_sanitizer', 'ci_job.xml.em', {
                 'cmake_build_type': 'Debug',
                 'time_trigger_spec': PERIODIC_JOB_SPEC,
-                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS + ' amazon-b9-ros@amazon.com',
```
Comment (thomas-moulard) [inline create_jenkins_job.py]: ditto

File: create_jenkins_job.py (hunk 2 above)
Comment (nuclearsandwich) [reply create_jenkins_job.py]: > is this email used somewhere else?

I pulled it off the [rcl_logging](https://github.com/ros2/rcl_logging/blob/efeebc970934ec0f9c7f64d0c175345074fa8ef7/rcl_logging_noop/package.xml#L8) maintainer entry.

**Comment by nuclearsandwich** [comment]
This is deployed and you can expect the first emails tonight.

**Pull request merged**